src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/failover/FailoverConnectionPlugin.java

```diff
@@ -143,28 +144,26 @@ public FailoverConnectionPlugin(
       Log logger) {
     this.currentConnectionProvider = currentConnectionProvider;
     this.propertySet = propertySet;
-    this.mainHost = this.currentConnectionProvider.getCurrentHostInfo();
     this.nextPlugin = nextPlugin;
     this.logger = logger;
-    this.initialConnectionProps = this.mainHost.getHostProperties();
-
-    initSettings();
-    this.connectionProvider = new BasicConnectionProvider();
 
-    if (!this.enableFailoverSetting) {
-      // Use a standard default connection - no further initialization required
-      try {
-        this.updateCurrentConnection(
-            this.connectionProvider.connect(this.mainHost),
-            this.mainHost);
-      } catch (SQLException e) {
-        // todo: proper handling
-        e.printStackTrace();
-      }
+    this.initialConnectionProps = new HashMap<>();
+    Properties originalProperties = this.propertySet.exposeAsProperties();
+    if (originalProperties != null) {
+      originalProperties.stringPropertyNames().stream()
+        .forEach(p -> this.initialConnectionProps.put(p, originalProperties.getProperty(p)));
+    }
 
-      return;
+    initSettings();
+    try {
+      initProxy();
+    } catch (SQLException e) {
+      // TODO: review
+      e.printStackTrace();
     }
```

Comment:
Why did we move initProxy up?

We need to initialize this.topologyService first